### PR TITLE
feat: ai-lead designer code quality refactor

### PR DESCRIPTION
**Pull request by PeterBaker0**
### Architecture + module boundaries

- Introduced clearer structure across domain/, store/, features/, integration/, and types/.
- Added/updated module docs and barrel exports for cleaner imports.

### Store and reducers

- Consolidated store usage around createDesignerStore.
- Centralized undo config in store/undoConfig.ts.
- Split monolithic uiSpec reducer into focused slices:
  - fieldReducers
  - sectionReducers
  - viewSetReducers
- Kept compatibility re-export via state/uiSpec-reducer.ts.

### Type safety + action usage

- Replaced remaining stringly-typed dispatch usage with typed action creators.
- Strengthened condition typing and extracted shared condition types.
- Improved condition control behavior for invalid operator/value states.

### Duplication reduction

- Added notebook/domain helpers (adapters, id helpers, field clone utilities).
- Added field editor registry and shared field update helper.
- Deduplicated host integration logic for Project/Template tabs via shared notebook adapter + save mutation hook.

### Legacy cleanup

- Removed retired standalone/legacy designer files (old app/store/localStorage/unused panels/schema/etc.).

### Tests

- Added/expanded targeted tests for:
  - condition domain utilities
  - field editor registry (including fallback rendering)
  - ui-spec reducer behavior coverage (field/section/viewset flows)
- Cleaned stale TODO/comment scaffolding in condition tests.

### CODE REVIEW

**Review by ranisa-gupta16 — COMMENTED** [review]
I've spent about 35–42 mins testing this PR manually.
AI is being used here as a big accelerator for large refactors, and this PR is a good example of that. I’ve done a decent stress test within that timefram e, and this looked strong. :-)

I did a bunch of field work in the designer (open/close editor for different field types + JSON changes + relcoads), including some rapid add/remove/reorder and undo,redo. Everything behaved normally for me, no major console warnings/errors. I tested field editors for different field types and did back and forth updates in JSON with different changes in data. It worked well for me and no crashes after doing it several times.

I feel performance-wise, agree with Peter that it’s performing well for me; especially with all that back and forth that I did. It wasn’t that bad; in the past it was a bit laggy sometimes on my system. With the time I’ve spent on this, anything major would’ve shown; this is really good. 

I tested the props and they are passing well in editor components. No weird UI glitches, and the accordion thing is fixed as well, thanks for giving it the right commands, Peter, reloaded the app several times with some different JSON being updated, it all synced well and I tested it back in app too, no issues found. 

I couldn’t exhaustively test every edge case manually, but I tested the key flows and they absolutely worked perfectly for me. 

I think AI did a great job, thanks for all the effort that went into this. :-)

File: web/src/designer/store/slices/uiSpec/fieldReducers.ts [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+      throw new Error(
+        `Cannot duplicate unknown field ${originalFieldName} via fieldDuplicated action`
+      );
+    }
+
+    const originalField = state.fields[originalFieldName];
+    const newField: FieldType = JSON.parse(JSON.stringify(originalField));
+    newField.designerIdentifier = uuidv4();
+
+    let fieldLabel = slugify(newFieldName);
```
Comment (stevecassidy) [inline web/src/designer/store/slices/uiSpec/fieldReducers.ts]: Again, use buildUniqueFieldName.

File: web/src/designer/store/slices/uiSpec/sectionReducers.ts [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+      condition: sourceSection.condition,
+    };
+
+    for (const originalFieldName of sourceSection.fields) {
+      if (!(originalFieldName in state.fields)) continue;
+      const originalField = state.fields[originalFieldName];
+      const baseLabel =
+        originalField['component-parameters'].label || originalFieldName;
+      const newFieldLabel = baseLabel;
+      let fieldSlug = slugify(newFieldLabel);
```
Comment (stevecassidy) [inline web/src/designer/store/slices/uiSpec/sectionReducers.ts]: Also here, buildUniqueFieldName
